src/components/ChannelPreview/ChannelPreview.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useState } from 'react';
+
+import {
+  ChannelPreviewCountOnly,
+  ChannelPreviewCountOnlyProps,
+} from './ChannelPreviewCountOnly';
+import {
+  getDisplayImage,
+  getDisplayTitle,
+  getLatestMessagePreview,
+} from './utils';
+
+import { ChatContextValue, useChatContext } from '../../context/ChatContext';
+import { useTranslationContext } from '../../context/TranslationContext';
+
+import type { Channel, Event, MessageResponse } from 'stream-chat';
+
+import type { ChannelPreviewCompactProps } from './ChannelPreviewCompact';
+import type { ChannelPreviewMessengerProps } from './ChannelPreviewMessenger';
+
+import type { AvatarProps } from '../Avatar/Avatar';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+export type ChannelPreviewUIComponentProps<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> =
+  | ChannelPreviewCompactProps<At, Ch, Co, Ev, Me, Re, Us>
+  | ChannelPreviewCountOnlyProps<At, Ch, Co, Ev, Me, Re, Us>
+  | ChannelPreviewMessengerProps<At, Ch, Co, Ev, Me, Re, Us>
+  | { lastMessage?: MessageResponse<At, Ch, Co, Me, Re, Us> };
+
+export type ChannelPreviewProps<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  /** Available from [chat context](https://getstream.github.io/stream-chat-react/#chat) */
```

Comment:
i know a paste, but i don't think this comment is accurate. i believe the channel prop is coming from the query channels response, as we're pulling the active channel out of the chat context via the hook

Reply:
yeah it comes from the return of the `channelRenderFilterFn` call or `usePaginatedChannels` if the other isn't present

Reply:
can you do a simple adjust for these? can always add more later but it's so wrong it makes me uneasy lol

Reply:
yeah